src/renderers/shared/hooks/ReactHostOperationHistoryHook.js

```diff
@@ -7,22 +7,33 @@
  * of patent rights can be found in the PATENTS file in the same directory.
  *
  * @providesModule ReactHostOperationHistoryHook
+ * @flow
  */
 
 'use strict';
 
-var history = [];
+import type { DebugID } from 'ReactInstanceType';
+
+export type Operation = {instanceID: DebugID} & (
+  {type: 'mount', payload: string} |
+  {type: 'insert child', payload: {toIndex: number, content: string}} |
+  {type: 'move child', payload: {fromIndex: number, toIndex: number}} |
+  {type: 'replace children', payload: string} |
+  {type: 'replace text', payload: string} |
+  {type: 'replace with', payload: string} |
+  {type: 'update styles', payload: any /* Style Object */} |
```

Comment:
I tend to think `mixed` is a safer default – either it will work forever or you will notice when you get a type error because it is not specific enough. You will already find new type errors when adding more Flow files that depend on inaccurate/incomplete types in existing files so using `mixed` isn't that different in this respect.